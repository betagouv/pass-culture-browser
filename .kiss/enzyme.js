import React from 'react'
import { shallow } from 'enzyme'

import MyComponent from '../MyComponent'

describe('src | components | MyComponent', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      const props = { errors: {} }
      const wrapper = shallow(<MyComponent {...props} />)
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
