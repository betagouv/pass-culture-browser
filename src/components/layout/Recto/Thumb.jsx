import PropTypes from 'prop-types'
import React from 'react'

const Thumb = ({ withMediation, src, translated }) => {
  const backgroundStyle = (src && { backgroundImage: `url('${src}')` }) || {}
  const thumbStyle = Object.assign(backgroundStyle, {
    backgroundSize: withMediation ? 'cover' : null,
    backgroundColor: 'grey',
  })
  return (
    <div className="thumb">
      {!withMediation && <div
        className="background"
        style={backgroundStyle}
                         />}
      <div
        className={`image ${translated !== undefined ? 'translatable' : ''} ${
          translated ? 'translated' : ''
        }`}
        style={thumbStyle}
      />
    </div>
  )
}

Thumb.defaultProps = {
  src: null,
  translated: false,
  withMediation: false,
}

Thumb.propTypes = {
  src: PropTypes.string,
  translated: PropTypes.bool,
  withMediation: PropTypes.bool,
}

export default Thumb
