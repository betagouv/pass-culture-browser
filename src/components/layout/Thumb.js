import PropTypes from 'prop-types'
import classnames from 'classnames'
import React from 'react'

const Thumb = ({ withMediation, src, translated }) => {
  const backgroundStyle = (src && { backgroundImage: `url('${src}')` }) || {}
  const thumbStyle = Object.assign(backgroundStyle, {
    backgroundSize: withMediation ? 'cover' : null,
  })
  return (
    <div className="thumb">
      {!withMediation && <div
        className="background"
        style={backgroundStyle}
                         />}
      <div
        className={classnames({
          image: true,
          translatable: translated !== undefined,
          translated,
        })}
        style={thumbStyle}
      />
    </div>
  )
}

Thumb.defaultProps = {
  src: null,
  translated: false,
  withMediation: null,
}

Thumb.propTypes = {
  src: PropTypes.string,
  translated: PropTypes.bool,
  withMediation: PropTypes.object,
}

export default Thumb
