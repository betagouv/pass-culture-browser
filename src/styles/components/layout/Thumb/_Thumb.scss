.thumb {
  height: 100%;
  width: 100%;
  position: relative;
  overflow: hidden;

  .background {
    filter: blur(20px);
    background-size: 100% auto;
    background-position: center;
    height: 100%;
    width: 100%;
    position: relative;
    background-repeat: no-repeat;
    box-sizing: content-box;
    padding: 50px;
    margin-top: -50px;
    margin-left: -50px;
    transition: all 0.25s ease-out;
  }

  .image {
    position: absolute;
    height: 100%;
    width: 100%;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    background-size: auto 100%;
    background-position: center;
    background-repeat: no-repeat;

    &.translatable {
      transition-duration: 250ms;
      transition-delay: 50ms;
      -webkit-transition-property: transform;
      transition-property: transform;
      will-change: transform;

      &.translated {
        transform: translate(0, -25%) scale(1.1);
      }
    }

  }
}
