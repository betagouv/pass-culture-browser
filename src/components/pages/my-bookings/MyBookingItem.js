/* eslint
  react/jsx-one-expression-per-line: 0 */
import PropTypes from 'prop-types'
import get from 'lodash.get'
import moment from 'moment'
import { capitalize, Icon } from 'pass-culture-shared'
import React from 'react'
import { connect } from 'react-redux'
import Dotdotdot from 'react-dotdotdot'
import { Link } from 'react-router-dom'

import Thumb from '../../layout/Thumb'
import Ribbon from '../../layout/Ribbon'
import { getQueryURL } from '../../../helpers'
import { THUMBS_URL } from '../../../utils/config'
import { getTimezone } from '../../../utils/timezone'
import { selectRecommendation } from '../../../selectors'

// TODO A tester
const getDateString = (date, tz) =>
  date &&
  capitalize(
    moment(date)
      .tz(tz)
      .format('dddd DD/MM/YYYY Ã  H:mm')
  )

const MyBookingItem = ({
  completedUrl,
  date,
  dateString,
  isCancelled,
  isEvent,
  mediationId,
  name,
  offerId,
  timezone,
  token,
}) => {
  const cssclass = (isEvent && 'event') || 'thing'
  const queryURL = getQueryURL({ mediationId, offerId })
  const linkURL = `/decouverte/${queryURL}/verso`
  const thumbUrl = `${THUMBS_URL}/mediations/${mediationId}`
  return (
    <li
      data-token={token}
      data-booked-date={date}
      data-booked-timezone={timezone}
      className={`booking-item mb16 ${cssclass}`}
    >
      {isCancelled && <Ribbon />}
      <Link to={linkURL}>
        <Thumb src={thumbUrl} />
        <div className="infos">
          <div className="top">
            <h5 title={name} className="fs18 is-semi-bold">
              <Dotdotdot clamp={date ? 2 : 3}>{name}</Dotdotdot>
            </h5>
            <span className="fs13">{dateString || 'permanent'}</span>
          </div>
          {!completedUrl && <div className="token">{token}</div>}
        </div>
        <div className="arrow">
          <Icon svg="ico-next-S" className="Suivant" />
        </div>
      </Link>
    </li>
  )
}

MyBookingItem.defaultProps = {
  completedUrl: null,
  date: null,
  dateString: null,
  isCancelled: false,
  isEvent: false,
  mediationId: null,
  name: null,
  offerId: null,
  token: null,
}

MyBookingItem.propTypes = {
  completedUrl: PropTypes.string,
  date: PropTypes.string,
  dateString: PropTypes.string,
  isCancelled: PropTypes.bool,
  isEvent: PropTypes.bool,
  mediationId: PropTypes.string,
  name: PropTypes.string,
  offerId: PropTypes.string,
  timezone: PropTypes.string.isRequired,
  token: PropTypes.string,
}

export default connect((state, ownProps) => {
  const { booking } = ownProps || {}
  const { isCancelled, recommendationId, stock, token } = booking
  const completedUrl = get(booking, 'completedUrl')
  const offerId = get(stock, 'resolvedOffer.id')
  const name = get(stock, 'resolvedOffer.eventOrThing.name')
  const date = get(stock, 'eventOccurrence.beginningDatetime')
  const departementCode = get(stock, 'resolvedOffer.venue.departementCode')
  const timezone = getTimezone(departementCode)
  const dateString = getDateString(date, timezone)

  const recommendation = selectRecommendation(state, recommendationId)
  const mediationId = get(recommendation, 'mediationId')

  const isEvent = Boolean(get(stock, 'resolvedOffer.eventId'))

  // injected
  return {
    completedUrl,
    date,
    dateString,
    isCancelled,
    isEvent,
    mediationId,
    name,
    offerId,
    recommendation,
    timezone,
    token,
  }
})(MyBookingItem)
