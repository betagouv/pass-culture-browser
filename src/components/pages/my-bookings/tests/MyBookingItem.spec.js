import { shallow } from 'enzyme'
import React from 'react'

import MyBookingItem from '../MyBookingItem'
import Thumb from '../../../layout/Thumb'

describe('src | components | MyBookingItem', () => {
  it('should match snapshot with required props', () => {
    // given
    const props = { timezone: 'Europe/Paris' }

    // when
    const wrapper = shallow(<MyBookingItem {...props} />)

    // then
    expect(wrapper).toBeDefined()
    expect(wrapper).toMatchSnapshot()
  })

  it('should render the recommendation thumb URL', () => {
    // given
    const props = {
      recommendation: {
        thumbUrl: 'https://example.net/mediation/image',
      },
      timezone: 'fr-FR',
    }

    // when
    const wrapper = shallow(<MyBookingItem {...props} />)

    // then
    expect(wrapper.find(Thumb).props().src).toEqual(
      'https://example.net/mediation/image'
    )
  })
})
