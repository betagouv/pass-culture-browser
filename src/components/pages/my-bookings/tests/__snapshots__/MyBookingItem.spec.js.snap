// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | MyBookingItem should match snapshot with required props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MyBookingItem
    completedUrl={null}
    date={null}
    dateString={null}
    isCancelled={false}
    isEvent={false}
    mediationId={null}
    name={null}
    offerId={null}
    recommendation={Object {}}
    timezone="Europe/Paris"
    token={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <Link
          to="/decouverte//verso"
        >
          <Thumb
            src={null}
            translated={false}
            withMediation={null}
          />
          <div
            className="infos"
          >
            <div
              className="top"
            >
              <h5
                className="fs18 is-semi-bold"
                title={null}
              >
                <Dotdotdot
                  clamp={3}
                  tagName="div"
                  truncationChar="…"
                  useNativeClamp={true}
                />
              </h5>
              <span
                className="fs13"
              >
                permanent
              </span>
            </div>
            <div
              className="token"
            />
          </div>
          <div
            className="arrow"
          >
            <Icon
              className="Suivant"
              svg="ico-next-S"
            />
          </div>
        </Link>,
      ],
      "className": "booking-item mb16 thing",
      "data-booked-date": null,
      "data-booked-timezone": "Europe/Paris",
      "data-token": null,
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Thumb
              src={null}
              translated={false}
              withMediation={null}
            />,
            <div
              className="infos"
            >
              <div
                className="top"
              >
                <h5
                  className="fs18 is-semi-bold"
                  title={null}
                >
                  <Dotdotdot
                    clamp={3}
                    tagName="div"
                    truncationChar="…"
                    useNativeClamp={true}
                  />
                </h5>
                <span
                  className="fs13"
                >
                  permanent
                </span>
              </div>
              <div
                className="token"
              />
            </div>,
            <div
              className="arrow"
            >
              <Icon
                className="Suivant"
                svg="ico-next-S"
              />
            </div>,
          ],
          "to": "/decouverte//verso",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "src": null,
              "translated": false,
              "withMediation": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="top"
                >
                  <h5
                    className="fs18 is-semi-bold"
                    title={null}
                  >
                    <Dotdotdot
                      clamp={3}
                      tagName="div"
                      truncationChar="…"
                      useNativeClamp={true}
                    />
                  </h5>
                  <span
                    className="fs13"
                  >
                    permanent
                  </span>
                </div>,
                <div
                  className="token"
                />,
              ],
              "className": "infos",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h5
                      className="fs18 is-semi-bold"
                      title={null}
                    >
                      <Dotdotdot
                        clamp={3}
                        tagName="div"
                        truncationChar="…"
                        useNativeClamp={true}
                      />
                    </h5>,
                    <span
                      className="fs13"
                    >
                      permanent
                    </span>,
                  ],
                  "className": "top",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <Dotdotdot
                        clamp={3}
                        tagName="div"
                        truncationChar="…"
                        useNativeClamp={true}
                      />,
                      "className": "fs18 is-semi-bold",
                      "title": null,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": null,
                        "clamp": 3,
                        "tagName": "div",
                        "truncationChar": "…",
                        "useNativeClamp": true,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": "h5",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "permanent",
                      "className": "fs13",
                    },
                    "ref": null,
                    "rendered": "permanent",
                    "type": "span",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": null,
                  "className": "token",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Icon
                className="Suivant"
                svg="ico-next-S"
              />,
              "className": "arrow",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "Suivant",
                "svg": "ico-next-S",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": [Function],
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <Link
            to="/decouverte//verso"
          >
            <Thumb
              src={null}
              translated={false}
              withMediation={null}
            />
            <div
              className="infos"
            >
              <div
                className="top"
              >
                <h5
                  className="fs18 is-semi-bold"
                  title={null}
                >
                  <Dotdotdot
                    clamp={3}
                    tagName="div"
                    truncationChar="…"
                    useNativeClamp={true}
                  />
                </h5>
                <span
                  className="fs13"
                >
                  permanent
                </span>
              </div>
              <div
                className="token"
              />
            </div>
            <div
              className="arrow"
            >
              <Icon
                className="Suivant"
                svg="ico-next-S"
              />
            </div>
          </Link>,
        ],
        "className": "booking-item mb16 thing",
        "data-booked-date": null,
        "data-booked-timezone": "Europe/Paris",
        "data-token": null,
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Thumb
                src={null}
                translated={false}
                withMediation={null}
              />,
              <div
                className="infos"
              >
                <div
                  className="top"
                >
                  <h5
                    className="fs18 is-semi-bold"
                    title={null}
                  >
                    <Dotdotdot
                      clamp={3}
                      tagName="div"
                      truncationChar="…"
                      useNativeClamp={true}
                    />
                  </h5>
                  <span
                    className="fs13"
                  >
                    permanent
                  </span>
                </div>
                <div
                  className="token"
                />
              </div>,
              <div
                className="arrow"
              >
                <Icon
                  className="Suivant"
                  svg="ico-next-S"
                />
              </div>,
            ],
            "to": "/decouverte//verso",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "src": null,
                "translated": false,
                "withMediation": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="top"
                  >
                    <h5
                      className="fs18 is-semi-bold"
                      title={null}
                    >
                      <Dotdotdot
                        clamp={3}
                        tagName="div"
                        truncationChar="…"
                        useNativeClamp={true}
                      />
                    </h5>
                    <span
                      className="fs13"
                    >
                      permanent
                    </span>
                  </div>,
                  <div
                    className="token"
                  />,
                ],
                "className": "infos",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <h5
                        className="fs18 is-semi-bold"
                        title={null}
                      >
                        <Dotdotdot
                          clamp={3}
                          tagName="div"
                          truncationChar="…"
                          useNativeClamp={true}
                        />
                      </h5>,
                      <span
                        className="fs13"
                      >
                        permanent
                      </span>,
                    ],
                    "className": "top",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <Dotdotdot
                          clamp={3}
                          tagName="div"
                          truncationChar="…"
                          useNativeClamp={true}
                        />,
                        "className": "fs18 is-semi-bold",
                        "title": null,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": null,
                          "clamp": 3,
                          "tagName": "div",
                          "truncationChar": "…",
                          "useNativeClamp": true,
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      "type": "h5",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "permanent",
                        "className": "fs13",
                      },
                      "ref": null,
                      "rendered": "permanent",
                      "type": "span",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": null,
                    "className": "token",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Icon
                  className="Suivant"
                  svg="ico-next-S"
                />,
                "className": "arrow",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "Suivant",
                  "svg": "ico-next-S",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": [Function],
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
